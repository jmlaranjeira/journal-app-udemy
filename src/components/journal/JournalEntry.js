

import React from 'react'

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
        <div className="journal__entry-picture"
        style={{
            backgroundSize: "cover",
            backgroundImage: "url(https://www.ngenespanol.com/wp-content/uploads/2018/08/La-primera-imagen-de-la-historia-1280x720.jpg)"
        }}></div>
        <div className="journal__entry-body">
            <p className="journal__entry-title">
                Un nuevo d√≠a
            </p>
            <p className="journal__entry-content">
                Reprehenderit anim occaecat officia nisi cupidatat deserunt. Consectetur ad nostrud voluptate aliquip ad aliquip. Reprehenderit mollit ea irure nulla do. Dolore elit occaecat ut qui ad velit voluptate ex.
            </p>
        </div>
        <div className="journal__entry-date-box">
            <span>Monday</span>
            <h4>28</h4>
        </div>
    </div>
  )
}
