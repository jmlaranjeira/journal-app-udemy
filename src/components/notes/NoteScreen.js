

import React from 'react'
import { NotesAppBar } from "./NotesAppBar"

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
        
        <NotesAppBar />

        <div className="notes__content">

            <input type="text" placeholder="Some awesome title" className="notes__title-input" autoComplete="off"/>
            <textarea placeholder="What happened today" className="notes__textarea"></textarea>
            <div className="notes__image">
                <img src="https://www.tooltyp.com/wp-content/uploads/2014/10/1900x920-8-beneficios-de-usar-imagenes-en-nuestros-sitios-web.jpg" alt="imagen" />
            </div>
        </div>
    </div>
  )
}
