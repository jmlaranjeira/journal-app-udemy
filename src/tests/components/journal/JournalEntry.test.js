import React from 'react'
import { mount } from "enzyme"
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from 'redux-mock-store' //ES6 modules
import '@testing-library/jest-dom';
import { JournalEntry } from "../../../components/journal/JournalEntry";
import { activeNote } from "../../../actions/notes";

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

let initState = {};

const note = {
    id: 10,
    date: 0,
    title: 'Hola',
    body: 'Mundo',
    url: 'https://prueba.com/imagen.png'
}

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={ store }>
        <JournalEntry {...note} />
    </Provider>
);

describe('Pruebas sobre el componente <JournalEntry />', () => {

    test('Debe de mostrarse el Snapshot correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de activar la nota', () => {

        wrapper.find('.journal__entry').prop('onClick')();
        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( note.id, {...note} )
        );
    })
})