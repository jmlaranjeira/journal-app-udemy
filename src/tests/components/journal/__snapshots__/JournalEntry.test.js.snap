// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <JournalEntry /> Debe de mostrarse el Snapshot correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://prueba.com/imagen.png)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      3rd
    </h4>
  </div>
</div>
`;
